UG: Fix bugs (#313)
AY2425S1-CS2103T-F08-2/tp: docs/UserGuide.md

  * Prefix: `[-t/DELETE-TAG]`
    - the tags specified will be **deleted** from the list of existing tags.
    - if the tag name you entered to delete **does not exist**, the edit will be successful but you will get the warning:
    - if the tag name you entered to delete **does not exist**, the edit will be successful, but you will get the warning:
      `You tried removing a nonexistent study group tag.`
* When editing **all other fields** : Existing values will be updated to the input values.

Examples:
*  `edit 1 a/29 e/johndoe@example.com` Edits the age and email address of the 1st person to be `29` and `johndoe@example.com` respectively.
*  `edit 2 n/Betsy Crower t/1B -t/2B` Edits the name of the 2nd person to be `Betsy Crower`, adds a new tag `1B` and deletes existing tag `2B`
*  `edit 2 n/Betsy Crower t/1B -t/2B` Edits the name of the 2nd person to be `Betsy Crower`, adds a new tag `1B` and deletes existing tag `2B`.

### Deleting persons : `delete`

Deletes the specified persons from the address book.
Deletes the specified person(s) from the address book.

Format: `delete [INDEX | INDEX-INDEX]…`

* Deletes the person at the specified `INDEX` or range of indices

> * The index must be within the range

Reviewer: Could you also add a point about requiring at least one optional field?

`* At least one of the index (range) must be provided.`

Format: `delete [INDEX | INDEX-INDEX]…`

> * At least one of the index (range) must be provided.

Reviewer 2: maybe it'll be clearer if its "At least one index or index range must be provided."?

> * At least one of the index (range) must be provided.

Reviewer 2: ```suggestion
* At least one index or index range must be provided.
```

* Each criterion can have more than one keyword. e.g., `n/alice bob hans`
* The search is case-insensitive. e.g., `n/hans` will match `Hans`
* The order of the keywords does not matter. e.g., `n/Hans Bo` will match `Bo Hans`
* Persons matching at least one keyword for **every specified criteria** will be returned.
  e.g., `n/Alice Bob g/f` will return `{Alice Wang, …, Female, …}` person but not `{Bob Tan, …, Male, …}` person
* Each criterion is matched based on the following:
  * Name, Detail, Study Group Tag: Only full words will be matched e.g. `n/Han` will not match `Hans`, and `d/assign` will not match `To be assigned`.
  * Email: Any substring will be matched e.g. `e/han` will match `hans@example.com`.

>   * Age: Only ages contained by the input age(s) and/or range(s) of ages will be matched e.g. `a/17 21-25` will match `17` and `25` but not `26`.

Reviewer 2: maybe use 18/19/20 instead of 26 cos i feel like a likely area of confusion would be numbers in between the 17 and the 21-25

>   * Age: Only ages contained by the input age(s) and/or range(s) of ages will be matched e.g. `a/17 21-25` will match `17` and `25` but not `26`.

Reviewer 2: ```suggestion
  * Age: Only ages contained by the input age(s) and/or range(s) of ages will be matched e.g. `a/17 21-25` will match `17` and `25` but not `19`.
```

## Command summary

Action | Format, Examples                                                                                                                                          
--------|-----------------------------------------------------------------------------------------------------------------------------------------------------------
**Help** | `help`                                                                                                                                                    
**Add** | `add n/NAME e/EMAIL g/GENDER a/AGE [d/DETAIL] [t/STUDY_GROUP_TAG]…`<br> e.g., `add n/James Ho e/jamesho@example.com g/M a/30 d/to be assigned  t/1A t/3C` 
**List** | `list`                                                                                                                                                    
**Edit** | `edit INDEX [n/NAME] [e/EMAIL] [g/GENDER] [a/AGE] [t/ADD_TAG]… [-t/DELETE_TAG]…`<br> e.g., `edit 2 n/James Lee e/jameslee@example.com t/1A t/1B`          

> **Delete** | `delete [INDEX \| INDEX-INDEX]…`<br> e.g., `delete 3 6 10-15 20 30-40`                                                                                    

Reviewer 2: what does this mean?

Author: Without the \, the code-format will not work properly! (It'll break at | instead)

Reviewer 2: ohh ok
